### DISCUSSION TITLE
Installing dev-tools

### DISCUSSION THREAD

**Discussion by depsypher** [excaliburjs/Excalibur#2740] [Q&A]
Anybody here know what this is about when I try to `npm install @excaliburjs/dev-tools`?

```
dyld[19625]: Library not loaded: /opt/homebrew/opt/icu4c/lib/libicui18n.70.dylib
  Referenced from: <B953AF6B-2CF4-344B-934F-A3116A459CC7> /opt/homebrew/Cellar/node@16/16.17.0/bin/node
  Reason: tried: '/opt/homebrew/opt/icu4c/lib/libicui18n.70.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/icu4c/lib/libicui18n.70.dylib' (no such file), '/opt/homebrew/opt/icu4c/lib/libicui18n.70.dylib' (no such file), '/usr/local/lib/libicui18n.70.dylib' (no such file), '/usr/lib/libicui18n.70.dylib' (no such file, not in dyld cache), '/opt/homebrew/Cellar/icu4c/73.2/lib/libicui18n.70.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/Cellar/icu4c/73.2/lib/libicui18n.70.dylib' (no such file), '/opt/homebrew/Cellar/icu4c/73.2/lib/libicui18n.70.dylib' (no such file), '/usr/local/lib/libicui18n.70.dylib' (no such file), '/usr/lib/libicui18n.70.dylib' (no such file, not in dyld cache)
zsh: abort      npm install @excaliburjs/dev-tools
```

**Comment by depsypher** [comment]
Hmm looks like something got hosed on my system in general, so almost certainly nothing to do with excalibur. Working through it. Will close this once I have a clearer picture.

**Comment by depsypher** [comment]
Ok, that was super weird, this solved that mysterious issue:
https://discuss.puredevops.com/t/library-not-loaded-usr-local-opt-icu4c-lib-libicui18n-69-dylib/179

Looks like I'm still having issues installing dev-tools though because of latest requires excalibur 0.28.0 and there seems to be a typo in this peer dependency within tiled?

```
ray@armada proto64 % npm install                              
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: sample-tiled-webpack@1.0.0
npm ERR! Found: excalibur@0.28.0
npm ERR! node_modules/excalibur
npm ERR!   excalibur@"0.28.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer excalibur@"~0.28.1" from @excaliburjs/plugin-tiled@0.28.0
npm ERR! node_modules/@excaliburjs/plugin-tiled
npm ERR!   @excaliburjs/plugin-tiled@"0.28.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/ray/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/ray/.npm/_logs/2023-08-15T18_02_48_909Z-debug-0.log
```

I'm guessing 0.28.1 in excalibur-tiled should really be 0.28.0?

**Reply by eonarheim** [reply]
Hi @depsypher thanks for the post, I'll look into this!

I have a working version of the latest versions in https://github.com/excaliburjs/sample-tiled-webpack

**Reply by eonarheim** [reply]
You might be around it for now by using `--legacy-peer-deps`

**Reply by eonarheim** [reply]
It's wild that you're getting a peer error. I could be missing something but this looks right to me the latest version of excalibur is v0.28.1, and both the [devtool](https://github.com/excaliburjs/dev-tools/blob/main/package.json#L31) and [tilled plugin](https://github.com/excaliburjs/excalibur-tiled/blob/main/package.json#L66) list v0.28.1 as their peer dependency.

**Reply by eonarheim** [reply]
I've tried a fresh install with `legacy-peer-deps=false` and it seems to work, what does your package.json look like?

<img width="359" alt="image" src="https://github.com/excaliburjs/Excalibur/assets/612071/c08252c0-be21-4911-a0ae-4f9a7577fcf6">
<img width="329" alt="image" src="https://github.com/excaliburjs/Excalibur/assets/612071/1fd300e2-a634-4a75-94e2-154188153857">

**Comment by depsypher** [comment]
Oh this is my bad. I was hand editing the package.json and really thought the latest excalibur was 0.28.0. Sorry to make you go hunting for nothing!

**Reply by eonarheim** [reply]
It's all good 😄 npm is mystifying to me, and it's easy to make a mistake in the peering stuff across repos. I appreciate the note!
